### ISSUE TITLE
Enum-style validation of "action" option blocks variable interpolation

### ISSUE THREAD

**Issue by magnusbaeck** [logstash-plugins/logstash-output-elasticsearch#312]
As evidenced by https://discuss.elastic.co/t/couchdb-plugin-set-dynamic-type-for-elasticsearch/2293/18,

```
Invalid setting for elasticsearch output plugin:

output {
elasticsearch {
# This setting must be a ["index", "delete", "create", "update"]
# Expected one of ["index", "delete", "create", "update"], got ["%{[@metadata][action]}"]
action => "%{[@metadata][action]}"
...
}
```

the static validation of the `action` option doesn't work with the dynamic variable interpolation that eventually will take place, but that code won't be reached because of the validation. I seems either the validation should be relaxed or calls to event.sprintf() should be dropped for `@action` to not give the impression that you actually can include %{name} variable references there.

**Comment by andrewvc** [comment]
Thanks for reporting this Magnus, I'll push out an update ASAP. 

Interestingly, even some rather old commits have this issue. It must have been broken a long time ago. I'll try and add a unit test for it as well.

**Comment by andrewvc** [comment]
Fixed in #313 will be released as 2.1.5 very soon. Thanks @magnusbaeck !
